ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/router_bgp.py

```diff
@@ -183,13 +206,21 @@ def _router_bgp_vrfs(self) -> dict | None:
                         else:
                             ip_address = self._mlag_peer_ibgp_ip
 
-                        bgp_vrf.setdefault("neighbors", {})[ip_address] = {"peer_group": self._peer_group_mlag_ipv4_underlay_peer_name}
+                        bgp_vrf.setdefault("neighbors", []).append({"ip_address": ip_address, "peer_group": self._peer_group_mlag_ipv4_underlay_peer_name})
 
-                address_families = {}
+                address_families = []
                 for bgp_peer in vrf["bgp_peers"]:
                     peer_ip = bgp_peer.pop("ip_address")
                     address_family = f"ipv{ipaddress.ip_address(peer_ip).version}"
-                    address_families.setdefault(address_family, {}).setdefault("neighbors", {})[peer_ip] = {"activate": True}
+                    flag = 0
+                    for family in address_families:
+                        if family["address_family"] == address_family:
+                            flag = 1
+                            family.setdefault("neighbors", []).append({"ip_address": peer_ip, "activate": True})
+                            break
+                    if not flag:
+                        address_families.append({"address_family": address_family, "neighbors": [{"ip_address": peer_ip, "activate": True}]})
+
```

Comment:
import `get_item` from plugin_utils.utils
```suggestion
                    neighbor = {"ip_address": peer_ip, "activate": True}
                    if (af := get_item(address_families, "address_family", address_family)) is None:
                        address_families.append({"address_family": address_family, "neighbors": [neighbor]})
                    else:
                      af["neighbors"].append(neighbor)

```